modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2498,6 +2510,8 @@ def binary_op(self, op, right_frame, join_type="outer"):
             Modin DataFrame to join with.
         join_type : str, default: "outer"
             Type of join to apply.
+        perform_column_reindex : bool, default: True
```

Comment:
I wonder if we can avoid introducing a new parameter to `binary_op` and compute a condition inside of the operator when `perform_reindex` is equal to `True` or `False` in order to pass it in to `_copartition`?

Reply:
I believe, we can avoid introducing the new parameter. Instead of this we just need to check amount of columns of `self`/`right_frame` of equality to `1` (this is only one case when `fillna` uses `binary_op`, @anmyachev correct me if I'm wrong) and make reindexing only by rows in this case.

Reply:
It'll be complex condition because there's another case where `binary_op` is used. For example, let's take `add` binary operation. Behavior of the function is different where type of left and right operands is `DataFrame` and amount of `self`/`right_frame` is `1`. In that case we need to do column `reindex`.

Code:
```python
import modin.pandas as pd
df1 = pd.DataFrame([1,2,3,4], columns=["a"])
df2 = pd.DataFrame([7,6,4,9], columns=["b"])
print(df1 + df2)
```

However we can't differentiate `fillna` and `add` cases in `binary_op` function because we can't check type of operands in `binary_op`. So I'm even not sure that is possible to define `perform_reindex` condition in `binary_op` layer.

Reply:
@prutskov @YarShev hope I answered you.

Reply:
I am still thinking about to avoid introducing the parameter. What if we do column reindex?

Reply:
If we do column reindex, then this simply will not allow us to correctly apply the fillna function. In the case when the series have different names, this will cause the right operand to consist entirely of nans (after reindex), which in the case of this function will simply do nothing with the left operand.

Reply:
I can try to enter a new value for parameter `join_type`. Something like: `left_only_rows`.

Reply:
@YarShev I managed to get rid of the new parameter in the `_copartition` function.